R/make-traceability-matrix.R

```diff
@@ -478,8 +483,17 @@ filter_symbol_functions <- function(funcs){
 }
 
 
-safe_expr <- function(expr){
-  tryCatch({
-    expr
-  }, error = function(e) e)
+safe_expr <- function(expr, verbose = FALSE){
+  tryCatch(
+    expr,
+    error = function(e){
+      if(isTRUE(verbose)) warning(warningCondition(e))
+      e
+    },
+    warning = function(w){
+      if(isTRUE(verbose)) warning(warningCondition(w))
+      w
+    },
+    message = function(m) m
```

Comment:
Please unlearn this pattern.  It will only bring unfortunate bugs.

 * https://github.com/metrumresearchgroup/mpn.scorecard/issues/11
 * f6b2efe (coverage: do not catch warnings when running covr, 2023-08-14)

Reply:
That shouldnt impact `covr` though? Don't see how that issue is relevant unless we ran this package through covr _and_ it triggered a warning

Reply:
> Don't see how that issue is relevant unless we ran this package through covr _and_ it triggered a warning

covr isn't a relevant piece.  See the demo in the issue I linked (gh-11).  `tryCatch(..., {message,warning} = ...)` is a bad pattern because it halts the execution on the first warning or message received.  

If you want to act on a message or warning, you can instead use `withCallingHandlers` and then resume execution.  Depending on the case, you may find it more convenient to use `purrr::quietly`, which uses `withCallingHandlers` underneath to handle messages and warnings and also has logic to capture standard output.